minetest/minetest
# Colorblindness correction support

Do not be mistaken by the name of this PR ("Colourblind correction"), this doesnt give colour blind people the ability to see colours as normal people can, for more information about what colour blind correction shaders do, see https://tylerdavidhoward.com/thesis/ and http://blog.noblemaster.com/2013/10/26/opengl-shader-to-correct-and-simulate-color-blindness-experimental/. These colour blind correction shaders would make minetest more inclusive to colour blind people.

For more information bout colour blindness in general, see https://www.colourblindawareness.org/

## To do

This PR is Ready for Review.


## How to test
(you cant really tell if it helps if you see colours normally, like me, but you can just trust the theory, which i do, and even ask someone colour blind that you know, which i did, and ask if it helps them, ofc **you will need to change the shader to the right one for their colour blindness**, btw, the person i asked said it helped them)
1) first, just open a singleplayer game(or server if you want) to see that normal colours still work
2) navigate to the settings tab and go to graphics->in-game->shaders and change the setting colour_blind_correction this is set by default to 0, meaning no colourblind correction, if changed to 1, then the protanopia correction shader would be enabled, if changed to 2, the deuteranopia shader would be enabled and finally, if changed to 3, the tritanopia shader would be enabled.
3) open a single player game now(or server if you want) and see the difference

## Screenshots

![image](https://user-images.githubusercontent.com/55553015/125681553-cc0a3f30-7e21-4e3f-b5a6-0fe8c89406b1.png)
![image](https://user-images.githubusercontent.com/55553015/125681702-72e0acef-6c23-4fb3-babf-45075d6dde9c.png)
![image](https://user-images.githubusercontent.com/55553015/125681824-f4d4e716-bcc7-4400-a3c8-9015de65b0e5.png)

## Credit

I got the functions that filter the normal colours to make it correct for colour blindness from https://www.shadertoy.com/view/XslyzX
